### ISSUE TITLE
Using Dataloader on a HeteroGraph and setting shuffle to False, the output_nodes are out of order.

### ISSUE THREAD

**Issue by maqy1995** [dmlc/dgl#4125]
DGL version: 0.8.2

code reproduce:
```python
import dgl
from ogb.nodeproppred import DglNodePropPredDataset

dataset = DglNodePropPredDataset(name="ogbn-mag", root='/mnt/disk1/dataset/gnn_dataset/dgl')
hetero_g, labels = dataset[0]
splitted_idx = dataset.get_idx_split()
train_nid = splitted_idx["train"]['paper']
val_nid = splitted_idx["valid"]['paper']
test_nid = splitted_idx["test"]['paper']

train_fanouts = []
for layer in range(2):
    train_fanouts.append({etype: 10 + layer for etype in hetero_g.canonical_etypes})

train_neighbor_sampler = dgl.dataloading.NeighborSampler(fanouts=train_fanouts)

train_dataloader = dgl.dataloading.DataLoader(
    graph=hetero_g,
    indices={'paper': train_nid},
    graph_sampler=train_neighbor_sampler,
    device="cpu",
    batch_size=100,
    shuffle=False,
    drop_last=False,
    num_workers=0)

train_iter = iter(train_dataloader)
_, first_seeds, _ = next(train_iter)

# should be 0,1,2...98,99 ?
print(first_seeds)

output:
{'paper': tensor([62, 74, 73, 72, 71, 70, 69, 68, 67, 66, 65, 64, 63, 75, 61, 60, 59, 58,
        57, 56, 55, 54, 53, 52, 51, 87, 99, 98, 97, 96, 95, 94, 93, 92, 91, 90,
        89, 88, 50, 86, 85, 84, 83, 82, 81, 80, 79, 78, 77, 76, 12, 24, 23, 22,
        21, 20, 19, 18, 17, 16, 15, 14, 13, 25, 11, 10,  9,  8,  7,  6,  5,  4,
         3,  2,  1, 37, 49, 48, 47, 46, 45, 44, 43, 42, 41, 40, 39, 38,  0, 36,
        35, 34, 33, 32, 31, 30, 29, 28, 27, 26])}
```

As we can see, the `first_seeds` is out of order, rather than a range from 0 to 100.

The reason for that seems to be `torch.argsort()` is not stable.
```python
class _TensorizedDatasetIter(object):

    def __next__(self):
        # convert the type-ID pairs to dictionary
        type_ids = batch[:, 0]
        indices = batch[:, 1]
        type_ids_sortidx = torch.argsort(type_ids) # not stable
```

@BarclayII, I'm not sure if this is a bug. `torch.sort(input,..., stable=True)` may work for this.

**Comment by mufeili** [comment]
Could you help open a PR and fix the issue?

**Comment by maqy1995** [comment]
Here is the pr: https://github.com/dmlc/dgl/pull/4147
